src/users/auth/auth.guard.ts

```diff
@@ -0,0 +1,12 @@
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
```

Comment:
what is this doing?

Reply:
this a guard should be added to the controller for authentication